backend/spec/requests/roles_request_spec.rb

```diff
@@ -2,14 +2,43 @@
 
 RSpec.describe 'RolesController', type: :request do
   describe '#list' do
-    let!(:role) { create(:role) }
+    let!(:role_moderator) { create(:role_moderator) }
+    let!(:role_teacher) { create(:role_teacher) }
+    let!(:role_student) { create(:role_student) }
+    let!(:role_admin) { create(:role_admin) }
+    let!(:not_admin_roles) do
+      [{ 'id' => role_moderator.id, 'role_type' => role_moderator.role_type },
+       { 'id' => role_teacher.id, 'role_type' => role_teacher.role_type }, { 'id' => role_student.id, 'role_type' => role_student.role_type }]
+    end
+    let!(:all_roles) { not_admin_roles }
+
+    let(:user_admin) { create(:user) }
+
+    describe 'when admin, list all roles' do
+      before do
+        get api_v1_roles_path, headers: auth_headers(user: user_admin)
+      end
+
+      it { expect(response).to have_http_status :ok }
+
+      it do
+        all_roles << { 'id' => role_admin.id, 'role_type' => role_admin.role_type }
```

Comment:
this should be done at line 13, otherwise the variable `all_roles` doesn't have all roles